### PR TITLE
Add autoscaling e2e test

### PR DESCRIPTION
**Pull request by enxebre**
As part of getting the test green the following fixes were needed:
- Bump autoscaler image to bring changes for the autoscaler to be able to restrict informers to a particular namespace so our rbac role is valid.
- Fixed creation timestamp check and unit tests for SetMachineDeploymentReplicas that was otherwise stepping on the replicas value set by the autoscaler.
- Change the Autolscaling.Min/Max fields to not be pointers since there's no need to diff between nil and 0.

### CODE REVIEW

**Comment by enxebre** [comment]
Need to dig into scaling down a little more since pods with local storage e.g grafana and image-registry and pos with anti-affinity rules make scaling down not reproducible.
/test e2e-aws-pooled

File: api/v1alpha1/nodepool_types.go [hunk 1]
```diff
@@ -150,10 +154,10 @@ type NodePoolManagement struct {
 type NodePoolAutoScaling struct {
 	// +optional
```
Comment (ironcladlou) [inline api/v1alpha1/nodepool_types.go:155]: Is this still +optional without the pointer?

File: api/v1alpha1/nodepool_types.go [hunk 2]
```diff
@@ -89,7 +89,11 @@ type NodePoolSpec struct {
 type NodePoolStatus struct {
 	// NodeCount is the most recently observed number of replicas.
 	// +optional
```
Comment (ironcladlou) [inline api/v1alpha1/nodepool_types.go:91]: Is this still +optional without the pointer?

File: api/v1alpha1/nodepool_types.go [hunk 3]
```diff
@@ -150,10 +154,10 @@ type NodePoolManagement struct {
 type NodePoolAutoScaling struct {
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	Min *int32 `json:"min"`
+	Min int32 `json:"min"`
 	// +optional
 	// +kubebuilder:validation:Minimum=1
-	Max *int32 `json:"max"`
+	Max int32 `json:"max"`
```
Comment (ironcladlou) [inline api/v1alpha1/nodepool_types.go:160]: Is this still +optional without the pointer?

File: test/e2e/scenarios/autoscaling.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+			Kind:       "Job",
+			APIVersion: "batch/v1",
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "autoscaling-workload",
+							Image: "busybox",
```
Comment (ironcladlou) [inline test/e2e/scenarios/autoscaling.go:177]: Is there a payload image we can use for this?

File: api/v1alpha1/nodepool_types.go (hunk 2 above)
Comment (enxebre) [reply api/v1alpha1/nodepool_types.go:91]: it was not a pointer, it's just changing from int to int32

File: api/v1alpha1/nodepool_types.go (hunk 1 above)
Comment (enxebre) [reply api/v1alpha1/nodepool_types.go:155]: this should actually be required when `NodePoolAutoScaling` is set, I'll update.

File: test/e2e/scenarios/autoscaling.go (hunk 4 above)
Comment (enxebre) [reply test/e2e/scenarios/autoscaling.go]: Updated to use the payload image.

**Comment by ironcladlou** [comment]
/lgtm
/hold

Great addition, thanks! Leaving a hold in case others wanted to chime in, feel free to remove when ready

**Comment by csrwng** [comment]
/hold cancel
lgtm

**Pull request merged**